main/acle.md

```diff
@@ -4932,6 +5044,1381 @@ assertions, which may speed up access to aligned data (and will fault
+  void sc_caller(void)
+  {
+    svint8_t i = ...;
+    svbool_t b = ...;
+
+    n_callee(i);   // Undefined behavior if sc_caller was called
+                   //   in streaming mode (but not otherwise)
+    s_callee(b);   // Undefined behavior if sc_caller was called
+                   //   in non-streaming mode (but not otherwise)
+    sc_callee(i);  // OK: keeps current streaming/non-streaming mode
+    ls_callee(b);  // Undefined behavior if sc_caller was called
+                   //   in streaming mode, but the call itself is
+                   //   OK otherwise. As noted above, the callee
+                   //   invokes undefined behavior internally
+  }
+```
+
+If the streaming and non-streaming vector lengths are equal then there
+is no undefined behavior in the code above.
+
+### `asm` restrictions related to streaming mode
+
+Some ACLE implementations might support the GNU “inline asm” extension.
+If so, the value of PSTATE.SM at the start of such inline asms is
+guaranteed to match the mode of the [abstract machine](#abstract-machine).
+For example, PSTATE.SM is guaranteed to be 1 at the start of inline asms
+in [streaming statements](#streaming-statement) and 0 at the start of
+inline asms in [non-streaming statements](#non-streaming-statement).
+
+All inline asms must preserve the value of PSTATE.SM: an asm entered
+in streaming mode must finish in streaming mode, and an asm entered
+in non-streaming mode must finish in non-streaming mode. The behavior
+in other cases is undefined.
+
+An inline asm can temporarily switch mode internally, such as using an
+SMSTART/SMSTOP pair. However, such mode switches invalidate all Z and P
+register state. The asm's “clobber list” must mention any registers
+whose values might be changed by the asm.
+
+## ZA storage
+
+### Introduction to ZA storage
+
+SME provides an area of storage called ZA, of size SVL.B×SVL.B bytes.
+It also provides a processor state bit called PSTATE.ZA to control
+whether ZA is enabled.
+
+<!-- Do not remove the following `span`, they are needed to create an
+anchor that can be referred via an internal hyperlink to the section
+following it. --><span id="za-state"></span>
+
+In C and C++ code, access to ZA is controlled at function granularity:
+a function either uses ZA or it does not. Another way to say this is
+that a function either “has ZA state” or it does not.
+
+If a function does have ZA state, the function can either share that ZA
+state with the function's caller or create new ZA state “from scratch”.
+In the latter case, it is the compiler's responsibility to free up ZA
+so that the function can use it; see the description of the lazy saving
+scheme in [[AAPCS64](#AAPCS64) for details about how the compiler
+does this.
+
+There is therefore a one-out-of-three choice for how a function handles ZA:
+
+1.  The function has no [ZA state](#za-state). This is the default.
+
+2.  The function has [ZA state](#za-state) that it shares with its caller.
```

Comment:
I don't mind changing it if there is strong preference for bullets, but the reason for using an enumerated list here is that the list is introduced by:

> These possibilities give a one-out-of-three choice for how a function handles ZA:

I think having 1., 2. and 3. next to the three choices is more natural in that context.  In prose it would probably have been “First(ly),”, “Second(ly),” etc.

Reply:
SGTM, leave them as numbered items.